subprojects/core/src/test/groovy/org/gradle/invocation/DefaultGradleSpec.groovy

```diff
@@ -423,9 +444,21 @@ class DefaultGradleSpec extends Specification {
         0 * action.execute(_)
 
         when:
-        gradle.buildListenerBroadcaster.projectsLoaded(gradle)
+        gradle.executeRootProjectActions()
+
+        then:
+        thrown(IllegalStateException)
```

Comment:
💅 Given the exception is rather generic, it's nice to assert the message as well